src/QirRuntime/test/QIR-static/qsharp/qir-test-conditionals.qs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+namespace Microsoft.Quantum.Testing.QIR {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+
+    operation TestApplyIf() : Unit {
+        use q1 = Qubit();
+        use q2 = Qubit();
+
+        // Zero branch
+        let r1 = M(q1);
+        ApplyIfElseRCA(r1, (X, q1), (Y, q1));
+        Adjoint ApplyIfElseRCA(r1, (X, q1), (Y, q1));
+        Controlled ApplyIfElseRCA([q2], (r1, (X, q1), (Y, q1)));
+        X(q2);
+        Adjoint Controlled ApplyIfElseRCA([q2], (r1, (X, q1), (Y, q1)));
+
+        // One branch
+        let r2 = M(q2);
+        ApplyIfElseRCA(r2, (Y, q1), (X, q1));
+
+        // Other variants
+        ApplyIfElseR(r1, (X, q1), (Y, q1));
+        Adjoint ApplyIfElseRA(r1, (X, q1), (Y, q1));
+        Controlled ApplyIfElseRC([q2], (r1, (X, q1), (Y, q1)));
+
+        ApplyIfOne(r2, (X, q1));
+        Adjoint ApplyIfOneA(r2, (X, q1));
+        Controlled ApplyIfOneC([q2], (r2, (X, q1)));
+        Adjoint Controlled ApplyIfOneCA([q2], (r2, (X, q1)));
+
+        ApplyIfZero(r1, (X, q1));
+        Adjoint ApplyIfZeroA(r1, (X, q1));
+        Controlled ApplyIfZeroC([q2], (r1, (X, q1)));
+        Adjoint Controlled ApplyIfZeroCA([q2], (r1, (X, q1)));
+    }
+
+    operation TestApplyConditionally() : Unit {
+        use q1 = Qubit();
+        use q2 = Qubit();
+
+        let r1 = M(q1);
+        X(q2);
+        let r2 = M(q2);
+
+        Microsoft.Quantum.Simulation.QuantumProcessor.Extensions.ApplyConditionally([r1], [r2], (Y, q1), (X, q1));
+        Microsoft.Quantum.Simulation.QuantumProcessor.Extensions.ApplyConditionally(
+            [r1, One], [Zero, r2], (X, q1), (Y, q1));
+
+        // Other variants
+        Adjoint Microsoft.Quantum.Simulation.QuantumProcessor.Extensions.ApplyConditionallyA(
+            [r1], [r2], (Y, q1), (X, q1));
+        Controlled Microsoft.Quantum.Simulation.QuantumProcessor.Extensions.ApplyConditionallyC(
+            [q2], ([r1], [r2], (Y, q1), (X, q1)));
+        Adjoint Controlled Microsoft.Quantum.Simulation.QuantumProcessor.Extensions.ApplyConditionallyCA(
+            [q2], ([r1], [r2], (Y, q1), (X, q1)));
+    }
+
+    operation TestConditionalRewrite() : Unit {

```

Comment:
Because the rest of the tests call into the conditional functions directly, this is the only test that depends on the rewrite step. But by changing the settings on the project in QIR-static/qsharp, now all if-statements will undergo a rewrite. I think it would be better to leave the existing project unchanged and include the test cases that explicitly call the ApplyIf variants, and then have a separate project that includes this test case that uses the execution target and suppressed C# generation as the place to test target dependent transformations.

Reply:
Agree. Removed the rewrite setting from the project and will only keep the direct tests for ApplyIf* and ApplyConditionally*